src/permutation/permutation.rs

```diff
@@ -838,18 +838,19 @@ mod test {
         perm.add_variables_to_map(var_zero, var_four, var_eight, var_nine, 3);
 
         /*
-        var_zero = {L0, R0,L1,L2, L3}
+        var_zero = {L0, R0, L1, L2, L3}
         var_two = {R1}
         var_three = {R2}
-        var_four = {R4}
-        var_five = {01}
-        var_six = {O2}
-        var_seven = {O3}
-        var_eight = {O4}
-        Left_sigma = {R0, L2,L3, L0}
+        var_four = {R3}
+        var_five = {O0}
+        var_six = {O1}
+        var_seven = {O2}
+        var_eight = {O3}
+        var_nine = {F0, F1, F2, F3}
+        Left_sigma = {R0, L2, L3, L0}
         Right_sigma = {L1, R1, R2, R3}
-        Out_sigma = {O0, O1, O2, O3, O4}
-        Fourth_sigma = {F0, F1, F2, F3, F4}
+        Out_sigma = {O0, O1, O2, O3}
+        Fourth_sigma = {F1, F2, F3, F0}
```

Comment:
The wire mappings here are associations of the gate outputs so the utility of an advice wire in this  context is as you write it but in some situations the scalar field. Fp, value we push may not always follow suite.

Reply:
Is there an exception in our code that not follow suit? I want to learn more.